pig-tests/test/org/partiql/pig/tests/SumConverterTests.kt

```diff
@@ -0,0 +1,97 @@
+package org.partiql.pig.tests
+
+import com.amazon.ionelement.api.ionBool
+import com.amazon.ionelement.api.ionInt
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.pig.tests.generated.ToyLang
+import kotlin.test.assertEquals
+
+/**
+ * Tests sum converters and and demonstrates their use to convert ToyLang's AST into a text representation
+ * that is vaguely similar to OCaml.  This example converts from a PIG domain sum type to a string, but these
+ * conversions can be applied to just about anything.
+ *
+ * ToyLang's AST actually has two sum types: expr and operator.
+ *
+ * This example also demonstrates how both can be used together.
+ */
+class SumConverterTests {
+val toyOperatorConverter = object : ToyLang.Operator.Converter<String> {
```

Comment:
nit (brought up offline): formatting of this class

Reply:
fixed.